lib/saml_idp/xml_security.rb

```diff
@@ -136,55 +139,82 @@ def validate_doc_params_signature(base64_cert, soft = true, params)
           sig_alg: params[:SigAlg]
         )
         log '***** validate_doc_params_signature: verify_signature:'
-        verify_signature(base64_cert, params[:SigAlg], Base64.decode64(params[:Signature]), canon_string, soft)
+        verify_signature(
+          base64_cert,
+          params[:SigAlg],
+          Base64.decode64(params[:Signature]),
+          canon_string, soft
+        )
       end
 
       def validate_doc_embedded_signature(base64_cert, soft = true)
         # check for inclusive namespaces
         inclusive_namespaces = extract_inclusive_namespaces
 
-        document = Nokogiri.parse(self.to_s)
+        document = Nokogiri.parse(to_s)
 
         # create a working copy so we don't modify the original
-        @working_copy ||= REXML::Document.new(self.to_s).root
+        @working_copy ||= REXML::Document.new(to_s).root
 
         # store and remove signature node
         @sig_element ||= begin
-                           element = REXML::XPath.first(@working_copy, "//ds:Signature", {"ds"=>DSIG})
-                           element.remove
-                         end
+          element = REXML::XPath.first(@working_copy, '//ds:Signature',
+                                       { 'ds' => DSIG })
```

Comment:
another "all same or all different"
```suggestion
          element = REXML::XPath.first(
            @working_copy,
            '//ds:Signature',
            { 'ds' => DSIG },
          )
```

Reply:
(this one was all rubocop)

Reply:
fixed in https://github.com/18F/saml_idp/pull/98/commits/2b8c1afa357e64a05ddf8932dfae209da2a40a6a